qualtran/bloqs/arithmetic/trigonometric.py

```diff
@@ -0,0 +1,110 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Dict, Set
+
+import numpy as np
+from attrs import frozen
+
+from qualtran import Bloq, bloq_example, BloqDocSpec, QFxp, Register, Signature
+from qualtran.bloqs.basic_gates import Toffoli
+from qualtran.bloqs.rotations.phase_gradient import _fxp
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+from qualtran.simulation.classical_sim import ClassicalValT
+from qualtran.symbolics import is_symbolic, SymbolicInt
+
+
+@frozen
+class ArcSin(Bloq):
+    r"""Compute the arcsine of a fixed-point number.
+
+    Implements the unitary:
+    $$
+        |a\rangle|0\rangle \rightarrow |a\rangle|\arcsin(a)\rangle
+    $$
+
+    Args:
+        bitsize: Number of bits used to represent the number.
+        num_frac: Number of fraction bits in the number.
+        num_iters: Number of Newton-Raphson iterations.
+        degree: Degree of the polynomial of the initial approximation.
+
+    Registers:
+        x: `bitsize`-sized input register.
+        result: `bitsize`-sized output register.
+
+    References:
+        [Optimizing Quantum Circuits for Arithmetic](https://arxiv.org/abs/1805.12445). Appendix D.
+    """
+
+    bitsize: SymbolicInt
+    num_frac: SymbolicInt
+    num_iters: SymbolicInt = 4  # reference studies 3, 4, or 5 iterations
+    degree: SymbolicInt = 4  # reference studies degree-3, 4, 5, or 6 polynomials
```

Comment:
can you move these comments (which might be missed or re-formatted) to the Args section of the docs? I.e. you can say "The default is 4 and the reference studies 3, 4, or 5 iterations"